src/components/badge/index.tsx

```diff
@@ -222,13 +236,27 @@ class Badge extends PureComponent<BadgeProps> {
         borderColor={borderColor}
         {...iconProps}
         style={{
-          flex: 1,
+          flex,
           ...iconStyle
         }}
       />
     );
   }
 
+  renderContent() {
+    const {withCounterIcon, icon, label} = this.props;
+
+    if (withCounterIcon && icon && label) {
```

Comment:
I think it's best to allow render both icon and label always (without an extra prop). 
It will be the user responsibility to pass either an icon, label or both.

TBH, not sure why we blocked it to begin with

Reply:
No problem, I tried to preserve the original logic 👍🏽